### PR TITLE
Bubble sort feature

### PR DESCRIPTION
**Pull request by Azeem838**
This pull request contains the following 2 methods:

1. bubble_sort takes an array and returns a sorted array
2. bubble_sort_by takes an array and a block and sorts the array based on the block.

Kindly find our commit below.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        array.insert(i + 1, array.delete_at(i))
+        i += 1
+      elsif (array[i] <=> array[i + 1]).zero?
+        i += 1
+      elsif (array[i] <=> array[i + 1]) == -1
+        i += 1
+      end
+    end
+  end
+  p array
```
Comment (uryelah) [inline bubble_sort.rb]: - [x] just return the 'array', not 'p array'

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+unsorted_list = [6, 2, 8, 1, 7, 3, 4]
```
Comment (uryelah) [inline bubble_sort.rb]: - [x] Please remove the debugging code before submiting for review. You should submit only the methods. (1)

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        i += 1
+      elsif (array[i] <=> array[i + 1]) == -1
+        i += 1
+      end
+    end
+  end
+  p array
+end
+
+bubble_sort(unsorted_list)
```
Comment (uryelah) [inline bubble_sort.rb]: - [x] Please remove the debugging code before submitting it for review. You should submit only the methods. (2)

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    (array.length - 1).times do
+      val = yield(array[i], array[i + 1])
+      array.insert(i + 1, array.delete_at(i)) if val.positive?
+      i += 1
+    end
+  end
+  p array
+end
+
+bubble_sort_by(%w[hi hello hey]) do |left, right|
```
Comment (uryelah) [inline bubble_sort.rb]: - [x] Please remove the debugging code before submitting it for review. You should submit only the methods. (1)

**Review by uryelah — CHANGES_REQUESTED** [review]
Hello **Abdoulaye Razak & Azeem Mahomed Ahmed**, I'm Sarah, your reviewer for this PR 👩‍🚀  

Great job guys, your methods pass all the required test cases!

I just can't approve you right away because you kept the debugging code in your file. Make sure to delete those before submitting, the file should have only the two methods.

--------

## **Changes Required**

**Check the comments below for details on the required changes:**

**Comment by uryelah** [comment]
**Kindly fix the pointed out issues before requesting the next code review**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team,  You have done a great job with this project. :clap:
#### You can go ahead and merge !!!

**Pull request merged**